cmd/admin-handlers.go

```diff
@@ -1768,3 +1773,141 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+	var wg sync.WaitGroup
+
+	switch madmin.InfoType(vars[madmin.INFO]) {
+	case madmin.ETCD:
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoints := globalEtcdClient.Endpoints()
+		endPoints := make([]string, len(etcdEndPoints))
+		errorString := make([]string, len(etcdEndPoints))
+
+		for i, endPoint := range etcdEndPoints {
+			wg.Add(1)
+			go func(i int, endPoint string) {
+				defer wg.Done()
+				endPoints[i] = endPoint
+				if err := checkConnection(endPoint); err != nil {
+					errorString[i] = errEtcdUnreachable.Error()
+				} else {
+					errorString[i] = ""
+				}
+
+			}(i, endPoint)
+		}
+		// Wait for all routines to finish.
+		wg.Wait()
+
+		etcdInfo := madmin.ETCDInfo{
+			ETCDEndpoint: endPoints,
+			Domain:       Environment.Get(EnvMinioDomain, ""),
+			PublicIPs:    globalDomainIPs,
+			Error:        errorString,
+		}
+		// Marshal API response
+		jsonBytes, err = json.Marshal(etcdInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+
+	case madmin.LOGGER:
+		var loggerInfo []madmin.ServerLoggerInfo
+		if endpoint, ok := Environment.Lookup(EnvLoggerHTTPEndpoint); ok {
+			result := madmin.ServerLoggerInfo{Endpoint: endpoint}
+			if e := checkConnection(endpoint); e != nil {
+				result.Error = e.Error()
+			}
+			loggerInfo = append(loggerInfo, result)
+
+		} else {
+			for _, http := range globalServerConfig.Logger.HTTP {
+				wg.Add(1)
+				go func(http loggerHTTP) {
+					defer wg.Done()
+					result := madmin.ServerLoggerInfo{Endpoint: http.Endpoint}
+					if http.Enabled {
+						e := checkConnection(http.Endpoint)
+						if e != nil {
+							result.Error = e.Error()
+						}
+					}
+					loggerInfo = append(loggerInfo, result)
+				}(http)
+			}
+			// Wait for all routines to finish.
+			wg.Wait()
+		}
+
+		// Marshal API response
+		jsonBytes, err = json.Marshal(loggerInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+
+	case madmin.AUDIT:
+		var auditInfo madmin.ServerAuditInfo
+		if endpoint, ok := Environment.Lookup(EnvAuditLoggerHTTPEndpoint); ok {
+			auditInfo = madmin.ServerAuditInfo{Endpoint: endpoint}
+			if e := checkConnection(endpoint); e != nil {
+				auditInfo.Error = e.Error()
+			}
+		}
+
+		// Marshal API response
+		jsonBytes, err = json.Marshal(auditInfo)
```

Comment:
this would always marshal an empty auditInfo

Reply:
We could have this 

```
        var auditInfo madmin.ServerAuditInfo
	
	endpoint, ok := Environment.Lookup(EnvAuditLoggerHTTPEndpoint)
	if !ok {
		return
	}
	
	auditInfo = madmin.ServerAuditInfo{Endpoint: endpoint}
	if e := checkConnection(endpoint); e != nil {			
			auditInfo.Error = e.Error()
	}
	
	// Marshal API response
	jsonBytes, err = json.Marshal(auditInfo)
	if err != nil {
		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
		return
	}
```

Reply:
done.